src/Xamarin.Android.Build.Tasks/Tasks/AssemblyModifierPipeline.cs

```diff
@@ -0,0 +1,248 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.JavaCallableWrappers;
+using Java.Interop.Tools.TypeNameMappings;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Framework;
+using Mono.Cecil;
+using MonoDroid.Tuner;
+using Xamarin.Android.Tools;
+using PackageNamingPolicyEnum = Java.Interop.Tools.TypeNameMappings.PackageNamingPolicy;
+
+namespace Xamarin.Android.Tasks;
+
+/// <summary>
+/// This task runs additional "linker steps" that are not part of ILLink. These steps
+/// are run *after* the linker has run. Additionally, this task is run by
+/// LinkAssembliesNoShrink to modify assemblies when ILLink is not used.
+/// </summary>
+public class AssemblyModifierPipeline : AndroidTask
+{
+	// Names of assemblies which don't have Mono.Android.dll references, or are framework assemblies, but which must
+	// be scanned for Java types.
+	static readonly HashSet<string> SpecialAssemblies = new HashSet<string> (StringComparer.OrdinalIgnoreCase) {
+			"Java.Interop.dll",
+			"Mono.Android.dll",
+			"Mono.Android.Runtime.dll",
+		};
```

Comment:
Does `XAJavaTypeScanner` also have a list of these?

https://github.com/dotnet/android/blob/599bd6c8fb3ffb1a0e09d06fb2b1fd69b8a57101/src/Xamarin.Android.Build.Tasks/Utilities/XAJavaTypeScanner.cs#L15-L23

Maybe they could share the list.

Reply:
That part of `XAJavaTypeScanner` is only used for scanning the old way for the steps that haven't been migrated.  It should go away once we complete the transition.